majordodo-core/src/main/java/majordodo/replication/ReplicatedCommitLog.java

```diff
@@ -648,11 +675,12 @@ public void recovery(LogSequenceNumber snapshotSequenceNumber, BiConsumer<LogSeq
                             if (LOGGER.isLoggable(Level.FINE)) {
                                 LOGGER.log(Level.FINE, "From entry {0}, to entry {1} ({2} %)", new Object[]{start, end, percent});                            
                             }
-                            Enumeration<LedgerEntry> seq = handle.readEntries(start, end);
-                            while (seq.hasMoreElements()) {
-                                LedgerEntry entry = seq.nextElement();
+                            
+                            Iterator<LedgerEntry> entries = handle.read(start, end).iterator();
```

Comment:
try (LedgerEntries entries = handle.read(start, end)) {
      for (LedgerEntry entry : entries) {
                  ..........
      }
}